skfs/src/FixedDir.sk

```diff
@@ -111,26 +122,155 @@ class FixedDir<T: frozen> private {
       keyComparison match {
       | LT() -> continue
       | GT() -> break false
-      | EQ() -> data[i - 1].value.i0.compare(data[i].value.i0)
+      | EQ() -> data[i - 1].source.compare(data[i].source)
       } match {
       | LT() -> continue
       | EQ() -> continue
       | GT() -> break false
       };
     };
     if (!sorted) {
-      data.sortBy(row ~> (row.key, row.value.i0));
+      data.sortBy(row ~> (row.key, row.source));
     };
     _ = static::computeTags(data, 0, data.size() - 1);
-    FixedDir{data => data.toArray()}
+    static{data => data.toArray()}
+  }
+
+  protected fun decode(row: FixedRow<T>): FixedRow<T> {
+    row
+  }
+}
+
+value class CompactRow<T>(
+  values: Array<T>,
```

Comment:
[for another diff(s)] With understanding of the schema and some table stats (which could be collected/updated during purge), we should be able to do a lot better than this. e.g. 

- we could do some simple flat representation encoding to pack values together in to a binary blob.
- we could track most common values and reference a dictionary (very good for highly repeated strings used as an enum, which is very common)
- nulls could be a small bitmask (most tables have few cols), and the values not represented at all
- etc

cc @mbouaziz - I'm sure you'll think of this stuff and more.